### PR TITLE
Adds framework for updating parameters via ROS

### PR DESCRIPTION
**Pull request by guyfleeman**

### CODE REVIEW

File: control-board/src/motion/robot_controller.rs [hunk 1]
```diff
@@ -229,4 +239,320 @@ impl<'a> BodyVelocityController<'a> {
...
+        }
+
+        if param_cmd.command_code == PCC_READ {
+            match param_cmd.parameter_name {
+                VEL_PID_X => {
+                    // set the type
+                    reply_cmd.data_format = PID_LIMITED_INTEGRAL_F32;
+
+                    // readback the data
+                    #[allow(non_snake_case)] // K is the mathematical symbol for this matrix
```
Comment (joe-spall) [inline control-board/src/motion/robot_controller.rs:283]: I am cool with changing the name to like pid_gain to get rid of these and add clarity

File: control-board/src/motion/robot_controller.rs [hunk 2]
```diff
@@ -229,4 +239,320 @@ impl<'a> BodyVelocityController<'a> {
...
+                },
+                _ => {
+                    defmt::debug!("unimplemented key write in RobotController");
+                    reply_cmd.command_code = PCC_NACK_INVALID_NAME;
+                    return Err(reply_cmd);
+                }
+            }
+        }
+
+        // match param_cmd.parameter_name {
```
Comment (joe-spall) [inline control-board/src/motion/robot_controller.rs:503]: Is this code WIP or can be deleted?

File: control-board/src/bin/control/control.rs [hunk 3]
```diff
@@ -499,10 +492,18 @@ impl<'a> Control<'a> {
                     self.back_left_motor.read_rads(),
                     self.back_right_motor.read_rads()
                 );
+
+                // currents current report as torques
+                let wheel_torques = Vector4::new(
```
[the remark this answers is not in the document]
Comment (guyfleeman) [reply control-board/src/bin/control/control.rs:497]: That conversion is done on the stspin using the df45 motor model. The modeled value is just getting stuffed in a packet variable called current. I've adjusted the comment and also marked it TODO for the naming/packet cleanup PR which should come after qualification

File: control-board/src/bin/control/radio.rs [hunk 4]
```diff
@@ -251,25 +179,36 @@ where
...
-                        {
-                            let mut last_control = last_control.lock().await;
-                            *last_control = true;
+                    let data_packet = radio.read_packet().await;
+                    if let Ok(data_packet) = data_packet {
+                        if let DataPacket::BasicControl(control) = data_packet {
+                            let mut latest_control = latest_control.lock().await;
+                            *latest_control = Some(control);
+
+                            // feed the disconnected watch-dog because we got a packet
```
[the remark this answers is not in the document]
Comment (guyfleeman) [reply control-board/src/bin/control/radio.rs:194]: I'm not sure what you mean? Perhaps this is a bit unclear, but this boolean flag is the watchdog. It's not using a timing critical hardware watchdog, a higher priority (preemptive) task periodically checks and clears this flag, so if the packet receiver doesn't set it, it shutdown motors.

File: control-board/src/motion/robot_controller.rs (hunk 2 above)
Comment (guyfleeman) [reply control-board/src/motion/robot_controller.rs:503]: Wanna test my rewrite above first on actual hardware, then it's all old/dead.

File: control-board/src/motion/robot_controller.rs (hunk 1 above)
Comment (guyfleeman) [reply control-board/src/motion/robot_controller.rs:283]: Kicking this one to #45 as it affects the CGKF as well and other bits of motion that need a refactor anyway

File: control-board/src/bin/hwtest-motor/main.rs [hunk 5]
```diff
@@ -105,7 +62,7 @@ async fn main(_spawner: embassy_executor::Spawner) {
     );
 
     let mut dotstar = Apa102::new(imu_spi);
-    dotstar.write([RGB8 { r: 10, g: 0, b: 0 }].iter().cloned());
+    let _res = dotstar.write([RGB8 { r: 10, g: 0, b: 0 }].iter().cloned());
```
[the remark this answers is not in the document]
Comment (guyfleeman) [reply control-board/src/bin/hwtest-motor/main.rs:65]: None in particular. Should we just do "_"? 

Semi-related, we need a better dotstar API

[the remark this answers is not in the document]
Comment (joe-spall) [reply control-board/src/bin/hwtest-motor/main.rs:65]: Yes I think so because putting "_" implies don't care better

**Comment by guyfleeman** [comment]
Latest commits address the review feedback, still need to test additional parameter updates on hardware. if it works, this can be merged

**Comment by hmmurdock** [comment]
All param read/writes confirmed on hardware. Landed the software-communications PR, and updated to main, re-tested a subset of params which also worked in hardware.

**Pull request merged**